agent/handlers/task_server_setup.go

```diff
@@ -56,9 +56,9 @@ func taskServerSetup(credentialsManager credentials.Manager,
 	containerInstanceArn string) *http.Server {
 	muxRouter := mux.NewRouter()
 
-	// Set this so that for request like "/v3//metadata/task", the Agent will pass
-	// it to task metadata handler instead of returning a 301 error.
-	muxRouter.SkipClean(true)
+	// Set this to false so that for request like "//v3//metadata/task"
+	// to permanently redirect(301) to "/v3/metadata/task" handler
+	muxRouter.SkipClean(false)
```

Comment:
Since this defaults to `false` (from the docs: "SkipClean defines the path cleaning behaviour for new routes. The initial value is false.") I initially wondered if we even need the explicit setting here.  
But I like it here.  It's deliberate.  
I would make the comment clearer (like the one in the docs: "When false, the path will be cleaned, so //v3//metadata/task will become /v3/metadata/task")

Reply:
I think a 301 redirect is more clear as it is not "becoming" the new url. it is a specific 301 redirect to the new url. let me know how you think

Reply:
I see ```w.WriteHeader(http.StatusMovedPermanently)``` aka 301.  Cool.